src/k8s/cmd/k8s/k8s_config.go

```diff
@@ -32,11 +32,7 @@ func newKubeConfigCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				return
 			}
 
-			if _, err := client.NodeStatus(cmd.Context()); err != nil {
-				cmd.PrintErrln("Error: The node is not part of a Kubernetes cluster. You can bootstrap a new cluster with:\n\n  sudo k8s bootstrap")
-				env.Exit(1)
-				return
-			}
+			_ = GetNodeStatus(client, cmd, env)
```

Comment:
This implementation hides a lot of details and is not particular easy to read. 
You would need to check the implementation of `GetNodeStatus` to understand what is going on and why we ignore everything here.

I would:
a) change the `GetNodeStatus` function to return the status and an error
b) create a new function e.g. `isBootstrapped` that wraps around the `GetNodeStatus` call, handles the error decoding logic and returns a bool
c) Then you can do something like this in the commands:
```
if isBootstrapped(...) {
   cmd.Print("Node already bootstrapped/part of a cluster etc")
   cmd.Exit(1)
}
```

Reply:
thanks for the amazing comment @bschimke95 . I tried to address your concern. it didn't turn out to be exactly like you suggested but I guess it's close. returning 3 things does bother me a bit but I feel it's not really that bad. it reduces duplication, it's easily testable and more readable than the previous version. the name `isBootstrapped` does not seem to work here since we need to return the `status` and an error as well.